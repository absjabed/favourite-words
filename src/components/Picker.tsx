import React from 'react';
import {StyleSheet} from 'react-native';
import RNPickerSelect, {PickerSelectProps} from 'react-native-picker-select';
import Icon  from 'react-native-vector-icons/FontAwesome5';
import Colors from '../constants/Colors/Colors';

const Picker: React.FC<PickerSelectProps> = (props: PickerSelectProps) => (
  <RNPickerSelect 
    {
      ...props
    }
    style={pickerSelectStyles}
    
    Icon={() => {
      return <Icon style={{left: 5, marginTop:15, color: Colors.white}} name="caret-down" />;
    }}
  >
  </RNPickerSelect>
);

export default Picker;

const pickerSelectStyles = StyleSheet.create({
  inputIOS: {
    fontSize: 16,
    paddingVertical: 12,
    paddingHorizontal: 10,
    borderWidth: 1,
    borderColor: 'gray',
    borderRadius: 4,
    color: 'black',
    paddingRight: 30, // to ensure the text is never behind the icon
    width: '100%',
  },
  inputAndroid: {
    width: '100%',
    fontSize: 14,
    marginLeft: 20,
    marginRight: 20,
    paddingHorizontal: 10,
    paddingVertical: 8,
    borderWidth: 0.5,
    borderColor: Colors.gray,
    borderRadius: 7,
    color: Colors.grey,
    fontFamily:'Ubuntu-Regular',
    paddingRight: 30, // to ensure the text is never behind the icon
  },
});