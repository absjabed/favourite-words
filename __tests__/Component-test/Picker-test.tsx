import 'react-native';
import React from 'react';
import Picker from '../../src/components/Picker';
import renderer, { create } from 'react-test-renderer';

  const placeholder = {
    label: 'Select type...',
    value: null,
    color: '#000',
  },
  favTypes = [{ label: "verb", value: "verb" },{ label: "noun", value: "noun" },{ label: "adverb", value: "adverb" }],
  mockOnPickerValueChanged = (value: any) => console.log("Picked value changed to", value),
  mockSelectedOption = "noun";

it('Picker renders correctly', () => {
  renderer.create(
              <Picker
                placeholder={placeholder}
                items={favTypes}
                onValueChange={(value) => mockOnPickerValueChanged(value)}
                value={mockSelectedOption}
                useNativeAndroidPickerStyle={false}
              />
              );
});

test('Picker snapshot', () => {
  const tree = create(
              <Picker
                placeholder={placeholder}
                items={favTypes}
                onValueChange={(value) => mockOnPickerValueChanged(value)}
                value={mockSelectedOption}
                useNativeAndroidPickerStyle={false}
              />
  );
  expect(tree).toMatchSnapshot();
});
