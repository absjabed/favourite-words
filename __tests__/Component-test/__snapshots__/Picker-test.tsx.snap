// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Picker snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignSelf": "stretch",
      },
      undefined,
    ]
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="ios_touchable_wrapper"
  >
    <View
      pointerEvents="box-only"
    >
      <TextInput
        allowFontScaling={true}
        editable={false}
        rejectResponderTermination={true}
        style={
          Array [
            Object {
              "borderColor": "gray",
              "borderRadius": 4,
              "borderWidth": 1,
              "color": "black",
              "fontSize": 16,
              "paddingHorizontal": 10,
              "paddingRight": 30,
              "paddingVertical": 12,
              "width": "100%",
            },
            Object {},
          ]
        }
        testID="text_input"
        underlineColorAndroid="transparent"
        value="noun"
      />
      <View
        style={
          Array [
            Object {
              "position": "absolute",
              "right": 0,
            },
            undefined,
          ]
        }
        testID="icon_container"
      >
        <Icon
          name="caret-down"
          style={
            Object {
              "color": "#ffffff",
              "left": 5,
              "marginTop": 15,
            }
          }
        />
      </View>
    </View>
  </View>
  <Modal
    hardwareAccelerated={false}
    onOrientationChange={[Function]}
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    testID="ios_modal"
    transparent={true}
    visible={false}
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flex": 1,
          "opacity": 1,
        }
      }
      testID="ios_modal_top"
    />
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#f8f8f8",
            "borderTopColor": "#dedede",
            "borderTopWidth": 1,
            "flexDirection": "row",
            "height": 45,
            "justifyContent": "space-between",
            "paddingHorizontal": 10,
            "zIndex": 2,
          },
          undefined,
        ]
      }
      testID="input_accessory_view"
    >
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            undefined,
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderColor": "#a1a1a1",
                  "borderRightWidth": 1.5,
                  "borderTopWidth": 1.5,
                  "height": 15,
                  "width": 15,
                },
                undefined,
                Object {
                  "marginLeft": 11,
                  "transform": Array [
                    Object {
                      "translateY": 4,
                    },
                    Object {
                      "rotate": "-45deg",
                    },
                  ],
                },
                undefined,
                Object {},
              ]
            }
          />
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderColor": "#a1a1a1",
                  "borderRightWidth": 1.5,
                  "borderTopWidth": 1.5,
                  "height": 15,
                  "width": 15,
                },
                undefined,
                Object {
                  "marginLeft": 22,
                  "transform": Array [
                    Object {
                      "translateY": -5,
                    },
                    Object {
                      "rotate": "135deg",
                    },
                  ],
                },
                undefined,
                Object {},
              ]
            }
          />
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        hitSlop={
          Object {
            "bottom": 4,
            "left": 4,
            "right": 4,
            "top": 4,
          }
        }
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="done_button"
      >
        <View
          testID="needed_for_touchable"
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "#007aff",
                  "fontSize": 17,
                  "fontWeight": "600",
                  "paddingRight": 11,
                  "paddingTop": 1,
                },
                undefined,
                Object {},
              ]
            }
            testID="done_text"
          >
            Done
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#d0d4da",
            "justifyContent": "center",
          },
          Object {
            "height": 215,
          },
          undefined,
        ]
      }
    >
      <View>
        <RNCPicker
          items={
            Array [
              Object {
                "label": "Select type...",
                "testID": undefined,
                "textColor": 4278190080,
                "value": null,
              },
              Object {
                "label": "verb",
                "testID": undefined,
                "textColor": undefined,
                "value": "verb",
              },
              Object {
                "label": "noun",
                "testID": undefined,
                "textColor": undefined,
                "value": "noun",
              },
              Object {
                "label": "adverb",
                "testID": undefined,
                "textColor": undefined,
                "value": "adverb",
              },
            ]
          }
          onChange={[Function]}
          selectedIndex={2}
          style={
            Array [
              Object {
                "height": 216,
              },
              undefined,
            ]
          }
          testID="ios_picker"
        />
      </View>
    </View>
  </Modal>
</View>
`;
